topcoffea/modules/datacard_tools.py

```diff
@@ -1016,13 +1029,15 @@ def decompose(self,h,wcs):
             r[quad_name] = {}
             for sparse_key in h._sumw.keys():
                 tup = tuple(x.name for x in sparse_key)
-                r[quad_name][tup] = 0.5*(tmp_lin_2[tup] - 2*tmp_lin_1[tup] + sm[tup])
-
+                r[quad_name][tup]=[]
+                for i in range(2):
+                    r[quad_name][tup].append( 0.5*(tmp_lin_2[tup][i] - 2*tmp_lin_1[tup][i] + sm[tup][i]) ) 
+                    
             for n2,wc2 in enumerate(wcs):
                 if n1 >= n2: continue
                 mixed_name = f"quad_mixed_{wc1}_{wc2}"
                 h.set_wilson_coefficients(**{wc1:1.0,wc2:1.0})
-                r[mixed_name] = h.values(overflow='all')
+                r[mixed_name] = h.values(overflow='all',sumw2=True)
                 terms += 1
```

Comment:
This is a general comment for the changes in this entire function. I _think_ it's safe to drop the `i` indexing for `numpy` arrays and it will automatically handle the multiplication and addition by sub-array. It also has the benefit of keeping everything in a  `numpy`/`awkward` array, rather than introducing a (admittedly small) vanilla python `list`.